File: TryPalindrome/TryPalindrome.cpp
#include <iostream>
#include <string>
#include <unordered_map>
#include <algorithm>

using namespace std;

int main() {
    string str;
    cin >> str;
    
    unordered_map<char, int> cnt;
    for (char c : str) {
        cnt[c]++;
    }
    
    int odd = 0;
    char odd_char = 0;
    for (pair<const char, int>P : cnt) {
        char c = P.first;
        int n = P.second;
        if (n & 1) {
            odd++;
            odd_char = c;
        }
    }
    
    if (odd > 1) {
        cout << "No\n";
        return 0;
    }
    
    string half, pal;
    for (pair<const char, int>P : cnt) {
        char c = P.first;
        int n = P.second;
        half += string(n / 2, c);
    }
    pal = half;
    if (odd) {
        pal += odd_char;
    }
    reverse(half.begin(), half.end());
    pal += half;
    cout << "Yes (\"" << pal << "\")\n";
    return 0;
}

File: TryPalindrome/TryPalindrome.js
const readline = require('readline');

const rl = readline.createInterface({
    input: process.stdin,
    output: process.stdout
});

rl.on('line', (s) => {
    s = s.trim();
    
    const cnt = {};
    for (const c of s) {
        cnt[c] = (cnt[c] || 0) + 1;
    }
    
    let odd = 0;
    let oddChar = null;
    for (const c in cnt) {
        if (cnt[c] % 2 === 1) {
            odd++;
            oddChar = c;
        }
    }
    
    if (odd > 1) {
        console.log("No");
        rl.close();
        return;
    }
    
    let half = "";
    for (const c in cnt) {
        half += c.repeat(Math.floor(cnt[c] / 2));
    }
    
    let pal = half;
    if (oddChar) {
        pal += oddChar;
    }
    pal += half.split('').reverse().join('');
    
    console.log(`Yes ("${pal}")`);
    rl.close();
});